### PR TITLE
test: Test for team reports page [7]

### PR DESCRIPTION
**Pull request by bistaastha**

### CODE REVIEW

**Review by mvaishnavi — APPROVED** [review]

File: src/app/fyle/team-reports/team-reports-1.page.spec.ts [hunk 1]
```diff
@@ -177,14 +182,20 @@ export function TestCases1(getTestBed) {
       it('should set filters in queryParams if filters is not defined in activatedRoute.snapshot.queryParams', () => {
```
Comment (Chethan-Fyle) [inline src/app/fyle/team-reports/team-reports-1.page.spec.ts:182]: Use `done()`.

File: src/app/fyle/team-reports/team-reports-1.page.spec.ts [hunk 2]
```diff
@@ -197,31 +208,19 @@ export function TestCases1(getTestBed) {
...
-          },
-          'example'
-        );
-      });
-
-      it('should call reportService.getTeamReports and update acc', () => {
-        apiV2Service.extendQueryParamsForTextSearch.and.returnValue({
-          rp_state: 'in.(APPROVER_PENDING)',
-        });
+      it('should call approverReporsService.getReportsByParams and update acc', () => {
```
Comment (Chethan-Fyle) [inline src/app/fyle/team-reports/team-reports-1.page.spec.ts:211]: Use `done()`.

File: src/app/fyle/team-reports/team-reports.page.ts [hunk 3]
```diff
@@ -172,9 +172,14 @@ export class TeamReportsPage implements OnInit {
 
       const paginatedPipe = this.loadData$.pipe(
         switchMap((params) => {
-          let queryParams = params.queryParams;
+          const queryParams = params.queryParams;
           const orderByParams = params.sortParam && params.sortDir ? `${params.sortParam}.${params.sortDir}` : null;
-          queryParams = this.apiV2Service.extendQueryParamsForTextSearch(queryParams, params.searchString);
+          if (params.searchString) {
+            queryParams.q = params.searchString;
+            queryParams.q = queryParams.q + ':*';
```
Comment (Chethan-Fyle) [inline src/app/fyle/team-reports/team-reports.page.ts:179]: Why need two lines here? Can't we just do `queryParams.q = params.searchString + ':*'`?

File: src/app/fyle/team-reports/team-reports.page.ts [hunk 4]
```diff
@@ -172,9 +172,14 @@ export class TeamReportsPage implements OnInit {
...
+          } else if (params.searchString === '') {
```
Comment (Chethan-Fyle) [inline src/app/fyle/team-reports/team-reports.page.ts:180]: Is this condition required? What if search string is undefined or null? Shouldn't we delete `q` in such cases?

File: src/app/fyle/team-reports/team-reports.page.ts [hunk 5]
```diff
@@ -197,8 +202,13 @@ export class TeamReportsPage implements OnInit {
...
         switchMap((params) => {
-          let queryParams = params.queryParams;
-          queryParams = this.apiV2Service.extendQueryParamsForTextSearch(queryParams, params.searchString);
+          const queryParams = params.queryParams;
+          if (params.searchString) {
+            queryParams.q = params.searchString;
+            queryParams.q = queryParams.q + ':*';
+          } else if (params.searchString === '') {
+            delete queryParams.q;
+          }
```
Comment (Chethan-Fyle) [inline src/app/fyle/team-reports/team-reports.page.ts:211]: Same as above.

**Review by Chethan-Fyle — APPROVED** [review]
Two minor comments.

**Pull request closed without merging**